compiler/src/scala/quoted/runtime/impl/printers/Extractors.scala

```diff
@@ -228,8 +228,9 @@ object Extractors {
       case PolyType(argNames, argBounds, resType) =>
         this += "PolyType(" ++= argNames += ", " ++= argBounds += ", " += resType += ")"
       case TypeLambda(argNames, argBounds, resType) =>
-        // resType is not printed to avoid cycles
-        this += "TypeLambda(" ++= argNames += ", " ++= argBounds += ", _)"
+        this += "TypeLambda(" ++= argNames += ", " ++= argBounds += ", " += resType += ")"
+      case ClassInfo(prefix, cls, parents, decls, selfInfo) =>
+        this += "ClassInfo(" += prefix += ", cls/*" += cls.fullName += "*/, " ++= parents += ", decls, " += selfInfo += ")"
```

Comment:
Is `cls` added by accident?

```suggestion
        this += "ClassInfo(" += prefix += ", /*" += cls.fullName += "*/, " ++= parents += ", decls, " += selfInfo += ")"
```

Reply:
No, we want to generate something like `ClassInfo(X, cls /*...*/, ..., ...)`. If we remove the `cls` it would not parse as valid scala code. The comment is there to hint at what is in that parameter but there is no structure to it.